### ISSUE TITLE
Better error message when accessing CNV data and some data is missing

### ISSUE THREAD

**Issue by alimanfoo** [malariagen/malariagen-data-python#363]
When trying to call functions like `ag3.gene_cnv('AGAP001356')`, if there are any releases where CNV data is not yet available, this will generate a cryptic error message:

```
FileNotFoundError: '.zmetadata'
```

Could be useful to either give a clearer error message, or just return data from the sample_sets that do have them.

**Comment by alimanfoo** [comment]
Yep this error will come and go, depending on the state of the data. If at any point there is an incomplete data release, where sample metadata and SNP data are present but CNV data are not present yet, you will hit this.

Also IIRC we may have a specific problem with Ag3.7 because some sample sets will never have CNV calls because they are outside the species we run CNV calling on.

**Comment by leehart** [comment]
@alimanfoo As mentioned in #555, it looks like if we skip samples without CNV HMM during `cnv_hmm()`, which is used by `_gene_cnv()` and indirectly by `gene_cnv_frequencies()`, then this will avoid the error messages, but it's not clear to me if there are any unintended implications, e.g. statistically or in plots, which might somehow be misleading.

```python
            lx = []
            for r in regions:
                ly = []
                for s in sample_sets:
                    y = self._cnv_hmm_dataset(
                        contig=r.contig,
                        sample_set=s,
                        inline_array=inline_array,
                        chunks=chunks,
                    )

                    # If no CNV HMM dataset was found then skip
                    if y is None:
                        continue

                    ly.append(y)

                debug("concatenate data from multiple sample sets")
                x = simple_xarray_concat(ly, dim=DIM_SAMPLE)

                debug("handle region, do this only once - optimisation")
                if r.start is not None or r.end is not None:
                    start = x["variant_position"].values
                    end = x["variant_end"].values
                    index = pd.IntervalIndex.from_arrays(start, end, closed="both")
                    # noinspection PyArgumentList
                    other = pd.Interval(r.start, r.end, closed="both")
                    loc_region = index.overlaps(other)  # type: ignore
                    x = x.isel(variants=loc_region)

                lx.append(x)
```

**Comment by leehart** [comment]
I believe this is resolved now. If there are still problems with the handling of missing CNV data, please open a new issue with a description.
